### PR TITLE
Fix `NaN` interpolation to catch edge cases and remove time dimension

### PR DESCRIPTION
**Pull request by ashjbarnes**
This fixes a few headaches in the initial condition interpolater. These are important for running with a newer executable where more netcdf variable type checks are done

### CODE REVIEW

**Comment by ashjbarnes** [comment]
Ok I ran black and then pushed again but the test fails and idk why

**Comment by navidcy** [comment]
Seems that the first commit you did black was not even able to run!
https://github.com/COSIMA/mom6-regional/actions/runs/5264807245/jobs/9516578361#step:3:23

There were 2 parentheses in line 635 of this commit? [254d848](https://github.com/COSIMA/mom6-regional/pull/35/commits/254d8486d7b4ba2e10bf20a319411534cf7d1d5d)

**Comment by navidcy** [comment]
That was a parse error -- I think not even python could parse that.

**Comment by navidcy** [comment]
OK! I think you have some WEIRD characters at the end of your file!

**Comment by navidcy** [comment]
I think it originated because you had a parsing error with 2 parentheses in your first commit....

File: mom6_regional/mom6_regional.py [hunk 1]
```diff
@@ -1503,3 +1457,4 @@ def brushcut(self, ryf=False):
             )
 
         return segment_out, encoding_dict
+  
```
Comment (navidcy) [inline mom6_regional/mom6_regional.py:1460]:
```suggestion

```

Comment (navidcy) [reply mom6_regional/mom6_regional.py]: There were some WEIRD characters here. If I opened with VS Code look what I saw:

Screenshot 2023-06-15 at 10 56 46 am

Comment (ashjbarnes) [reply mom6_regional/mom6_regional.py]: Yeah I saw that and thought I fixed it with the second commit. That's not present in my local copy (black actually ran fine for me) but maybe I forgot to save and commit again. Just re-committed now

**Comment by navidcy** [comment]
That's good...
But I'd have to trust you that you fixed a bug. This is the type of things that a test should have caught ;)

**Review by navidcy — APPROVED** [review]

**Comment by navidcy** [comment]
Gotcha!

**Comment by navidcy** [comment]
Note: If this was part of dealing with #34 then just feel free to commit there. :)

**Pull request merged**